packages/iris-grid/src/IrisGrid.jsx

```diff
@@ -1536,42 +1539,56 @@ export class IrisGrid extends Component {
   }
 
   freezeColumnByColumnName(columnName) {
-    const { frozenColumns } = this.state;
+    const { frozenColumns, movedColumns } = this.state;
     const { model } = this.props;
     log.debug2('freezing column', columnName);
 
-    let allFrozenColumns;
+    const allFrozenColumns =
+      frozenColumns == null
+        ? new Set(model.layoutHints?.frozenColumns)
+        : new Set(frozenColumns);
 
-    if (frozenColumns == null) {
-      allFrozenColumns = new Set(model.layoutHints?.frozenColumns ?? []);
-      allFrozenColumns.add(columnName);
-    } else {
-      allFrozenColumns = new Set([...frozenColumns, columnName]);
-    }
+    allFrozenColumns.add(columnName);
+
+    const modelIndex = model.getColumnIndexByName(columnName);
+    const visibleIndex = GridUtils.getVisibleIndex(modelIndex, movedColumns);
+    const newMovedColumns = GridUtils.moveItem(
+      visibleIndex,
+      allFrozenColumns.size - 1,
+      movedColumns
+    );
 
     this.setState({
       frozenColumns: [...allFrozenColumns],
+      movedColumns: newMovedColumns,
     });
   }
 
   unFreezeColumnByColumnName(columnName) {
-    const { frozenColumns } = this.state;
+    const { frozenColumns, movedColumns } = this.state;
     const { model } = this.props;
     log.debug2('unfreezing column', columnName);
 
-    let allFrozenColumns;
+    const allFrozenColumns =
+      frozenColumns == null
+        ? new Set(model.layoutHints?.frozenColumns)
```

Comment:
Should just be `model.frozenColumns`, as the result is then memoized:
```suggestion
        ? new Set(model.frozenColumns)
```

Reply:
Good catch. This is how the previous version accessed the model frozen columns.

I also switched the same call in freeze by name ~25 lines up from this one